src/com/google/javascript/jscomp/parsing/parser/Parser.java

```diff
@@ -2824,7 +2824,7 @@ private ParseTree parseAssignment(Expression expressionIn) {
 
     if (isStartOfAsyncArrowFunction(left)) {
       // re-evaluate as an async arrow function.
-      resetScanner(left);
+      resetScanner(start);
```

Comment:
Ah, I see.
Looks like I was incorrect in my mental model when I wrote `resetScanner()`.

In that case, please remove the version of `resetScanner()` that takes a `ParseTree`.
Thanks.